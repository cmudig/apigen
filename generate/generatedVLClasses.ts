
class Mark {
    constructor(private type: "bar" | "area" | "line" | { type: "bar" | "area" | "line"}) {}
    
  }
  
export function mark(type: "bar" | "area" | "line" | { type: "bar" | "area" | "line"}){
    return new Mark(type);
  }

class X {
    constructor(private field: string, private type: "quantitative" | "ordinal") {}
    
  }
  
export function x(field: string, type: "quantitative" | "ordinal"){
    return new X(field, type);
  }

class Y {
    constructor(private field: string, private type: "quantitative" | "ordinal") {}
    
  }
  
export function y(field: string, type: "quantitative" | "ordinal"){
    return new Y(field, type);
  }

class Color {
    constructor(private field: string, private type: "quantitative" | "ordinal") {}
    
  }
  
export function color(field: string, type: "quantitative" | "ordinal"){
    return new Color(field, type);
  }

class Encoding {
    constructor(private x?: X, private y?: Y, private color?: Color) {}
    
  }
  
export function encode(x?: X, y?: Y, color?: Color){
    return new Encoding(x, y, color);
  }

class Spec {
    constructor(private mark: Mark, private data: string, private encode: Encoding) {}
    
  }
  
export function spec(mark: Mark, data: string, encode: Encoding){
    return new Spec(mark, data, encode);
  }

export function toSpec(obj: any){
      return obj;
    }
export function toJSON(obj: any){
      return JSON.stringify(obj);
    }