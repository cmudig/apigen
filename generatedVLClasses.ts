import {} from './utils'

class Mark {
    constructor(private type: "bar" | "area" | "line") {}
    
}
  
export function mark(type: "bar" | "area" | "line"){
    return new Mark(type);
}

class Spec {
    constructor(private mark: Mark) {}
    
}
  
export function spec(mark: Mark){
    return new Spec(mark);
}

export function toSpec(obj: any){
    return JSON.stringify(obj);
}
