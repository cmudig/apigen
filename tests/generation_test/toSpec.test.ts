import * as VL from "../../generatedVLClasses";

/**
 * Test the generated file is working.
 */
describe('markSpec', () => {

    const spec = VL.spec(VL.mark("bar"));

    it('test toSpec()', () => {
        expect(VL.toSpec(spec)).toEqual("\{\"mark\":{\"type\":\"bar\"}}");
    });

    it('test toSpec() MatchSnapShot', () => {
        expect(VL.toSpec(spec)).toMatchSnapshot();
    });

});