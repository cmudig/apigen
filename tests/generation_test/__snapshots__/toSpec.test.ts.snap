// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markSpec test toSpec() MatchSnapShot 1`] = `"{"mark":{"type":"bar"}}"`;
